source/javascripts/controllers/_MainController.js.erb

```diff
@@ -338,16 +349,48 @@ import _ from "underscore";
 				}
 			};
 
+			viewModel.openUpdateYmlInRepositoryModal = function() {
+				var variables = appService.getCurrentVariables();
+				appService.validateVariables(variables)
+					.then(function() {
+							viewModel.showUpdateYmlInRepositoryModal = true;
+					})
+			}
```

Comment:
This should only be done in the Workflows/Env vars/Triggers tab. The bitrise.yml tab needs a separate one, which only has to show the modal, no validation is required. It might even fail, if the page is initially loaded with the bitrise.yml tab, because in that case there will be no `appService.appConfig`.